ffi/diplomat/src/provider.rs

```diff
@@ -24,30 +38,44 @@ pub mod ffi {
         pub success: bool,
     }
 
-    fn convert_provider<D: BufferProvider + 'static>(x: D) -> Box<ICU4XDataProvider> {
-        Box::new(ICU4XDataProvider(Box::new(x)))
+    #[cfg(feature = "any_provider")]
+    fn convert_any_provider<D: icu_provider::AnyProvider + 'static>(
```

Comment:
suggestion (nb, no preference): `new_any()` ?

Reply:
It used to be `convert_provider` so I'll just keep that nomenclature. What we really want is an `Into` but it was hard to get that working right with all the features and bounds.

Reply:
wfm